### PR TITLE
New `date` macro

### LINKED ISSUE #5427: Cross-database `date` macro
## Docs Changes Needed from dbt-core Issue #8831

At a minimum, update body to include a link to the page on docs.getdbt.com requiring updates and what part(s) of the page you would like to see updated.

### Background context

- https://github.com/dbt-labs/dbt-adapters/issues/192
- https://github.com/dbt-labs/dbt-core/issues/10075

### Page requiring updates

https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros

### What part(s) of the page to update

Add documentation for the new cross-database `date(year, month, day)` macro:
- description
- function signature / parameters
- code example
- rendered example

See https://github.com/dbt-labs/docs.getdbt.com/pull/5278 for an example.

### PR DESCRIPTION
**Pull request by nghi-ly**
## What are you changing in this pull request and why?

For dbt Core 1.8, add new `date` macro to the page https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros

closes #5427 

## Checklist
- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [x] Needs review from DX team

### CODE REVIEW

File: website/docs/reference/dbt-jinja-functions/cross-database-macros.md [hunk 1]
```diff
@@ -884,6 +887,29 @@ For databases that support it, this macro will return `NULL` when the cast fails
...
+ * `day`: an integer
+
+This macro converts an ISO 8601-formatted date string into an SQL `DATE` type.
+ 
+**Usage**:
+
+
+**Sample output**:
+
+
```
Comment (nghi-ly) [inline website/docs/reference/dbt-jinja-functions/cross-database-macros.md:910]: @dbeatty10 : is there a doc or something where i could grab info for the Usage and Sample output parts? unless i totally missed it, i wasn't able to find any from the issue nor the epic

**Comment by nghi-ly** [comment]
@dbeatty10 : could you review this for technical accuracy? i'm not sure i've updated the page correctly for `date`. thank you so much!

File: website/docs/reference/dbt-jinja-functions/cross-database-macros.md [hunk 2]
```diff
@@ -884,6 +887,35 @@ For databases that support it, this macro will return `NULL` when the cast fails
...
+ * `year`: an integer
+ * `month`: an integer
+ * `day`: an integer
+
+This macro converts the `year`, `month`, and `day` into an SQL `DATE` type.
+ 
+**Usage**:
+
+```sql
+{{ dbt.to_date(2023, 10, 4) }}
```
Comment (dbeatty10) [inline website/docs/reference/dbt-jinja-functions/cross-database-macros.md]:
```suggestion
{{ dbt.date(2023, 10, 4) }}
```

File: website/docs/reference/dbt-jinja-functions/cross-database-macros.md [hunk 3]
```diff
@@ -884,6 +887,35 @@ For databases that support it, this macro will return `NULL` when the cast fails
...
+```
+
+**Sample output (PostgreSQL)**:
+
+```sql
+2023-10-04
```
Comment (dbeatty10) [inline website/docs/reference/dbt-jinja-functions/cross-database-macros.md]:
```suggestion
to_date('2023-10-04', 'YYYY-MM-DD')
```

**Review by dbeatty10 — APPROVED** [review]
Looks good -- thanks @nghi-ly 🤩

File: website/docs/reference/dbt-jinja-functions/cross-database-macros.md [hunk 4]
```diff
@@ -884,6 +887,35 @@ For databases that support it, this macro will return `NULL` when the cast fails
...
+
+<VersionBlock firstVersion="1.8">
+
+__Args__:
+
+ * `year`: an integer
+ * `month`: an integer
+ * `day`: an integer
+
+This macro converts the `year`, `month`, and `day` into an SQL `DATE` type.
```
Comment (matthewshaver) [inline website/docs/reference/dbt-jinja-functions/cross-database-macros.md:903]:
```suggestion
This macro converts the `year`, `month`, and `day` into a SQL `DATE` type.
```

**Review by matthewshaver — APPROVED** [review]

File: website/docs/reference/dbt-jinja-functions/cross-database-macros.md (hunk 4 above)
Comment (nghi-ly) [reply website/docs/reference/dbt-jinja-functions/cross-database-macros.md:903]: hmmm, i was taught this -- non blocker so merging

Screenshot 2024-05-09 at 1 42 08 PM

**Pull request merged**